src/FEValues/face_values.jl

```diff
@@ -186,6 +100,52 @@ Return the normal at the quadrature point `qp` for the active face of the
 """
 getnormal(fv::FaceValues, qp::Int) = fv.normals[qp]
 
+nfaces(fv::FaceValues) = length(fv.geo_values)
+
+function checkface(fv::FaceValues, face::Int)
+    0 < face <= nfaces(fv) || error("Face index out of range.")
+    return nothing
+end
+
+function reinit!(fv::FaceValues, x::AbstractVector{Vec{dim,T}}, face_nr::Int, cell=nothing) where {dim, T}
+    check_reinit_sdim_consistency(:FaceValues, shape_gradient_type(fv), eltype(x))
+    @boundscheck checkface(fv, face_nr)
+    n_geom_basefuncs = getngeobasefunctions(fv)
+    length(x) == n_geom_basefuncs || throw_incompatible_coord_length(length(x), n_geom_basefuncs)
+    
+    fv.current_face[] = face_nr
+
+    geo_values = get_geo_values(fv)
+    fun_values = get_fun_values(fv)
+    @inbounds for (q_point, w) in pairs(getweights(fv.qr, face_nr))
+        mapping = calculate_mapping(geo_values, q_point, x)
+        J = getjacobian(mapping)
+        weight_norm = weighted_normal(J, getrefshape(geo_values.ip), face_nr)
+        detJ = norm(weight_norm)
```

Comment:
Maybe we can leave a reference to the `embedded_det` docs here, because they do basically the same thing.